### PR TITLE
ReceiveQueueListener_class_111701031

### PR DESCRIPTION
**Pull request by Drisya-Ponmari**
Class which dequeues from the receiving queue and calls INotification handler method by passing message.
Added comments and logging

### CODE REVIEW

File: src/ReceiveQueueListener.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+     * @return Nothing
+     * @exception exception    Error when unable to call onCallback method.
+     * @see Exception
+     */
+
+    @Override
+    public void run() {
+       
+        
+        while(true){
```
Comment (sunny117) [inline src/ReceiveQueueListener.java]: Use LanCommunicator.isRunning for while condition to make it consistent with other listeners.

**Review by sunny117 — CHANGES_REQUESTED** [review]

File: src/ReceiveQueueListener.java (hunk 1 above)
Comment (sravand123) [reply src/ReceiveQueueListener.java]: Thanks you @sunny117  for pointing it out

Comment (sravand123) [reply src/ReceiveQueueListener.java]: @Drisya-Ponmari  please update this

Comment (sravand123) [reply src/ReceiveQueueListener.java]: @Drisya-Ponmari  and also please keep this file in the folder src/networking/

File: src/ReceiveQueueListener.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    
+    /**
+     * Receiving queue of the processing module or content module based.
+     */
+    private IQueue<IncomingPacket> receivequeue;
+    
+    /**
+     * Logger instance for logging errors and activities.
+     */
+    ILogger logger;
```
Comment (sravand123) [inline src/ReceiveQueueListener.java]: get logger from LoggerFactory
ILogger logger = LoggerFactory.getLoggerInstance()

File: src/ReceiveQueueListener.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+         * Obtaining the module ( either processing or content module )handler from the handlerMap corresponding the identifier.
+         */
+        INotificationHandler handler = handlerMap.get(identifier);
+        
+        /**
+         * Calling the module handler method by passing the message.
+         */
+        if ( handler != null ){
+            
+            handler.onMessageRecieved(message);
```
Comment (sravand123) [inline src/ReceiveQueueListener.java]: Change received spelling

**Review by sravand123 — CHANGES_REQUESTED** [review]

File: src/ReceiveQueueListener.java (hunk 1 above)
Comment (Drisya-Ponmari) [reply src/ReceiveQueueListener.java]: > Use LanCommunicator.isRunning for while condition to make it consistent with other listeners.

Yeah correct. Thanks

Comment (Drisya-Ponmari) [reply src/ReceiveQueueListener.java]: > @Drisya-Ponmari and also please keep this file in the folder src/networking/

Okay

File: src/ReceiveQueueListener.java (hunk 2 above)
Comment (Drisya-Ponmari) [reply src/ReceiveQueueListener.java]: Okay

File: src/ReceiveQueueListener.java (hunk 3 above)
Comment (Drisya-Ponmari) [reply src/ReceiveQueueListener.java]: Okay .

**Review by sravand123 — CHANGES_REQUESTED** [review]
change folder name from network to networking

File: src/ReceiveQueueListener.java (hunk 1 above)
Comment (sravand123) [reply src/ReceiveQueueListener.java]: For a change use Lancommunicator.getStatus( ) for isRunning

File: src/network/ReceiveQueueListener.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        
+        while( LanCommunicator.isRunning ){
+            
+            if( !receivequeue.isEmpty() ){
+            
+                /**
+                 * Incoming packet which contains message and identifier is removed from the front of the queue if queue is empty.
+                 */
+                IncomingPacket packet = receivequeue.front();
+
```
Comment (sravand123) [inline src/network/ReceiveQueueListener.java]: Please dequeue the packet here

**Review by sravand123 — CHANGES_REQUESTED** [review]

File: src/network/ReceiveQueueListener.java (hunk 4 above)
Comment (Drisya-Ponmari) [reply src/network/ReceiveQueueListener.java]: Thanks for pointing this important bug

**Review by sravand123 — APPROVED** [review]
All good !!...good work drisya...

**Comment by sravand123** [comment]
[RecieveQueueListener](https://trello.com/c/ZZCC8Qwj/3-recievequeuelistener)

**Pull request merged**